src/freertos_drivers/arduino/ArduinoFs.hxx

```diff
@@ -0,0 +1,304 @@
+/** \copyright
+ * Copyright (c) 2020, Balazs Racz
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file ArduinoFs.hxx
+ *
+ * Single-file POSIX compatible filesystem using Arduino's EEPROM
+ * implementation for STM32.
+ *
+ * This file needs to be included into the .ino of the arduino sketch. It may
+ * only be compiled once as it has definitions of C library functions.
+ *
+ * @author Balazs Racz
+ * @date 18 July 2020
+ */
+
+#ifndef _FREERTOS_DRIVERS_ARDUINO_ARDUINOFS_HXX_
+#define _FREERTOS_DRIVERS_ARDUINO_ARDUINOFS_HXX_
+
+#ifndef ARDUINO_ARCH_STM32
+#error This module only works for STM32 boards.
+#endif
+
+#ifndef OPENMRN_HAVE_POSIX_FD
+#error You must add a file build_opt.h to the sketch directory and add -DOPENMRN_HAVE_POSIX_FD to it to make use of this module.
+#endif
+
+#include "stm32_eeprom.h"
+#include <errno.h>
+
+#define EEPROM_FILENAME "/dev/eeprom"
+
+/// Class that holds information and code for the single-file filesystem for
+/// emulating eeprom.
+class FsStatic
+{
+public:
+    /// Number of possible open file desriptors.
+    static constexpr unsigned MAX_FD = 8;
+    /// Offset markng that a file descriptor is not in use.
+    static constexpr unsigned UNUSED_FILE = (off_t)-1;
```

Comment:
would the datatype be better as off_t since it is only ever used in that type?

Reply:
done